server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/meta/StarTreeMetaWriter.java

```diff
@@ -0,0 +1,163 @@
+import java.util.List;
+
+import static org.opensearch.index.compositeindex.CompositeIndexConstants.COMPOSITE_FIELD_MARKER;
+import static org.opensearch.index.compositeindex.datacube.startree.fileformats.StarTreeWriter.VERSION_CURRENT;
+
+/**
+ * The utility class for serializing the metadata of a star-tree data structure.
+ * The metadata includes information about the dimensions, metrics, and other relevant details
+ * related to the star tree.
+ *
+ * @opensearch.experimental
+ */
+public class StarTreeMetaWriter {
+
+    private static final Logger logger = LogManager.getLogger(StarTreeMetaWriter.class);
+
+    /**
+     * Writes the star-tree metadata.
+     *
+     * @param metaOut                the IndexOutput to write the metadata
+     * @param starTreeField          the star-tree field
+     * @param metricAggregatorInfos  the list of metric aggregator information
+     * @param segmentAggregatedCount the aggregated document count for the segment
+     * @param numNodes               number of nodes in the star tree
+     * @param dataFilePointer        the file pointer to the start of the star tree data
+     * @param dataFileLength         the length of the star tree data file
+     * @throws IOException if an I/O error occurs while serializing the metadata
+     */
+    public static void writeStarTreeMetadata(
+        IndexOutput metaOut,
+        StarTreeField starTreeField,
+        List<MetricAggregatorInfo> metricAggregatorInfos,
+        Integer numNodes,
+        Integer segmentAggregatedCount,
+        long dataFilePointer,
+        long dataFileLength
+    ) throws IOException {
+
+        long initialMetaFilePointer = metaOut.getFilePointer();
+
+        writeMetaHeader(metaOut, CompositeMappedFieldType.CompositeFieldType.STAR_TREE, starTreeField.getName());
+        writeMeta(metaOut, metricAggregatorInfos, starTreeField, numNodes, segmentAggregatedCount, dataFilePointer, dataFileLength);
+
+        logger.debug(
+            "Star tree meta size in bytes : {} for star-tree field {}",
+            metaOut.getFilePointer() - initialMetaFilePointer,
+            starTreeField.getName()
+        );
+    }
+
+    /**
+     * Writes the star-tree metadata header.
+     *
+     * @param metaOut            the IndexOutput to write the header
+     * @param compositeFieldType the composite field type of the star-tree field
+     * @param starTreeFieldName  the name of the star-tree field
+     * @throws IOException if an I/O error occurs while writing the header
+     */
+    private static void writeMetaHeader(
+        IndexOutput metaOut,
+        CompositeMappedFieldType.CompositeFieldType compositeFieldType,
+        String starTreeFieldName
+    ) throws IOException {
+        // magic marker for sanity
+        metaOut.writeLong(COMPOSITE_FIELD_MARKER);
+
+        // version
+        metaOut.writeVInt(VERSION_CURRENT);
+
+        // star tree field name
+        metaOut.writeString(starTreeFieldName);
```

Comment:
Isn't this actual meta rather than header?

Reply:
Composite Field Name and Composite Field Type would be common across all future composite fields. Hence, before reading the actual contents of the metadata, we can read these two attributes.

If you see the implementation [here](https://github.com/opensearch-project/OpenSearch/pull/14809/files#diff-233e42a9e1b8258ede88cffccd6a4a906cd0e34172d243966b5878357f5fe66fR135), we parse the name and type of composite field first, before moving on to the field type specific reads.

Reply:
Also, header is an internal construct. Storage wise if we store it in header or meta, it won't matter. 
At a file level, we still rely on index codec headers

Reply:
nit : We can probably move the name and type to meta though.

Reply:
> Also, this header is an internal construct. Storage wise if we store it in header or meta, it won't matter.
At a file level, we still rely on index codec headers

Yes, the comment was from readability perspective